pkg/kubelet/kubelet_pods.go

```diff
@@ -557,18 +558,26 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string) (map[string]string, error) {
 // Make the environment variables for a pod in the given namespace.
 func (kl *Kubelet) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container, podIP string) ([]kubecontainer.EnvVar, error) {
 	var result []kubecontainer.EnvVar
+	var serviceEnv map[string]string
+	var err error
 	// Note:  These are added to the docker Config, but are not included in the checksum computed
 	// by kubecontainer.HashContainer(...).  That way, we can still determine whether an
 	// v1.Container is already running by its hash. (We don't want to restart a container just
 	// because some service changed.)
 	//
+	// Making the environment variables with services is optional and legacy. It's an old method
+	// of accessing services, associated with the deprecated --link option in Docker. Nowadays,
+	// DNS should be used instead of env vars - which is the default behavior.
+	//
 	// Note that there is a race between Kubelet seeing the pod and kubelet seeing the service.
 	// To avoid this users can: (1) wait between starting a service and starting; or (2) detect
 	// missing service env var and exit and be restarted; or (3) use DNS instead of env vars
 	// and keep trying to resolve the DNS name of the service (recommended).
-	serviceEnv, err := kl.getServiceEnvVarMap(pod.Namespace)
-	if err != nil {
-		return result, err
+	if _, ok := pod.Annotations[kubeapi.ServiceEnvironmentVariablesKey]; ok {
```

Comment:
This should be a field, not an annotation.

Reply:
done